traitlets/config/application.py

```diff
@@ -826,6 +828,40 @@ def exit(self, exit_status=0):
         self.log.debug("Exiting application: %s" % self.name)
         sys.exit(exit_status)
 
+    def write_config_file(self):
```

Comment:
@krinsman can we just make `config_file` a parameter of `Application.write_config_file()`? Subclasses can then pass in a `config_file` attribute if they have them.

Reply:
@rmorshea That makes sense to me. I don't know how the JupyterHub developers feel about it though, so I should refrain from speaking for them.

I probably don't have time in the near future to fix this PR, so if you come up with a better solution that makes this PR obsolete, please feel free to close it!